schema.py

```diff
@@ -66,8 +66,9 @@ class And(object):
     """
     def __init__(self, *args, **kw):
         self._args = args
-        assert list(kw) in (['error'], [])
+        assert sorted(list(kw)) in (['error', 'schema'], ['error'], [])
         self._error = kw.get('error')
+        self._schema = kw.get('schema', Schema)
```

Comment:
Hmm, what is the rationale for this? Could you add a short comment explaining why someone can pass the class here? Why shouldn't every subclass do `self.__class__` instead?